tests/python/pants_test/pantsd/test_pailgun_server.py

```diff
@@ -78,6 +79,46 @@ def test_process_request_thread_error(self, mock_shutdown_request):
     self.assertIs(self.mock_handler_inst.handle_error.called, True)
     mock_shutdown_request.assert_called_once_with(self.server, mock_request)
 
+  def test_process_request_thread_waits_until_timeout(self):
```

Comment:
Please run this test at least 20 times locally to increase the likelihood of not introducing a flake. Both with `./pants` and `./pants2`. 

The uses of `time.sleep()` look legitimate to me, but it is a huge culprit for flaky tests.

Reply:
Running
```
for i in {1..100}; do ./pants test tests/python/pants_test/pantsd:pailgun_server -- -k test_process_request_thread_waits_until_timeout 2> /dev/null; echo $? ; done
```
Now... it's going to take a while, but I agree that it's worth it. Will run with `./pants2` later

Reply:
Both `./pants` and `./pants2` seem to succeed 100% of the time in 100 runs each.